htdocs/compta/facture/class/facture.class.php

```diff
@@ -810,7 +814,37 @@ public function create(User $user, $notrigger = 0, $forceduedate = 0)
 							$newinvoiceline->fk_remise_except = $discountId;
 						}
 
-						$result = $newinvoiceline->insert();
+						$result = $this->addline(
```

Comment:
The future is to abandon the method addline that is stored into object and instead use the method   ->insert of the objectline.
So old code seems nearer the target.
Can you try your fix keeping the "->insert" ?

Reply:
Hello
What about all the calculation made in addline ?

Reply:
Hum, right.
So no choice, let's go with addline().